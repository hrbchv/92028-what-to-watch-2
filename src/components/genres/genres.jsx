import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreators} from "../../reducer/reducer";

class Genres extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      genres: props.genres,
      currentGenre: props.currentGenre,
      onChangeGenreClick: props.onChangeGenreClick
    };
    this._onChangeGenreClick = this._onChangeGenreClick.bind(this);
  }

  _onChangeGenreClick(event) {
    event.preventDefault();
    this.state.onChangeGenreClick(event.target.text);
  }

  render() {
    const {genres, currentGenre} = this.props;
    return <ul className="catalog__genres-list">
      {genres.map((genre, index) => {
        return <li
          key={genre + index}
          className={genre === currentGenre ? `catalog__genres-item catalog__genres-item--active` : `catalog__genres-item`}>
          <a onClick={this._onChangeGenreClick} href="#" className="catalog__genres-link">{genre}</a>
        </li>;
      })}
    </ul>;
  }
}

Genres.propTypes = {
  genres: PropTypes.array.isRequired,
  currentGenre: PropTypes.string.isRequired,
  onChangeGenreClick: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  currentGenre: state.currentGenre,
  genres: state.genres
});

const mapDispatchToProps = (dispatch) => {
  return {
    onChangeGenreClick: (value) => {
      dispatch(ActionCreators.changeGenre(value));
    }
  };
};

export {Genres};
export default connect(mapStateToProps, mapDispatchToProps)(Genres);
