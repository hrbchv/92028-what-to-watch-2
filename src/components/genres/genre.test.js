import React from "react";
import Enzyme, {shallow} from "enzyme";
import {Genres} from "./genres.jsx";
import Adapter from "enzyme-adapter-react-16/build/index";

Enzyme.configure({adapter: new Adapter()});

const filmsGenres = [`Romance`, `Sci-Fi`, `Thrillers`];
const currentGenre = `Thrillers`;
const onChangeGenreClick = jest.fn();

describe(`Genre`, () => {
  it(`should renders correctly`, () => {
    const tree = shallow(<Genres
      genres={filmsGenres}
      currentGenre={currentGenre}
      onChangeGenreClick={onChangeGenreClick}/>);
    expect(tree).toMatchSnapshot();
  });
});

