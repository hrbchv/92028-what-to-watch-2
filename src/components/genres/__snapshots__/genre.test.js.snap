// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre should renders correctly 1`] = `ShallowWrapper {}`;
